### ISSUE TITLE
My npm package is broken after v19 update

### ISSUE THREAD

**Issue by quedicesebas** [angular/angular#58811]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

After updating to Angular 19, using official documentation and `ng update` command, my angular library published on nm stops working (the js module is not found: Cannot find module 'ngx-bottom-sheet-modal' or its corresponding type declarations.)

### Please provide a link to a minimal reproduction of the bug

[Package in use](https://stackblitz.com/edit/ngx-bottom-sheet-modal-demo) (you can switch `ngx-bottom-sheet-modal` package from 4.01 to 3.01 version and will work) 

[Library repo](https://github.com/quedicesebas/angular-tailwind-ui/)

### Please provide the exception or error you saw

```true
Cannot find module 'ngx-bottom-sheet-modal' or its corresponding type declarations.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 19.0.1
Node: 20.15.1
Package Manager: npm 10.8.2
OS: win32 x64

Angular: 19.0.0
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1900.1
@angular-devkit/build-angular   19.0.1
@angular-devkit/core            19.0.1
@angular-devkit/schematics      19.0.1
@angular/cli                    19.0.1
@angular/ssr                    19.0.1
@schematics/angular             19.0.1
ng-packagr                      19.0.0
rxjs                            7.8.1
typescript                      5.6.3
zone.js                         0.15.0
```

**Comment by JeanMeche** [comment]
Hi, we will need a narrowed down repro project that exhibit this issue (either a github repo or a stackblitz demo). 
Without this, this isn't actionnable to us.

**Comment by quedicesebas** [comment]
> Hi, we will need a narrowed down repro project that exhibit this issue (either a github repo or a stackblitz demo). Without this, this isn't actionnable to us.

DONE

**Comment by JeanMeche** [comment]
It looks like your library isn't bundled correctly, the package.json is missing the right entries.

**Comment by quedicesebas** [comment]
> It looks like your library isn't bundled correctly, the package.json is missing the right entries.

Can you elaborate? I just migrate versions, before that it was working. I also just tried to create a project from scratch and publish it again (v4.0.2), but the same error appears.

**Comment by JeanMeche** [comment]
`4.0.1` was missing 

```
 "module": "fesm2022/ngx-bottom-sheet-modal.mjs",
  "typings": "index.d.ts",
  "exports": {
    "./package.json": {
      "default": "./package.json"
    },
    ".": {
      "types": "./index.d.ts",
      "default": "./fesm2022/ngx-bottom-sheet-modal.mjs"
    }
  }
```

This is now fixed in `4.0.2` and you project works fine.

**Comment by quedicesebas** [comment]
> `4.0.1` was missing
> 
> ```
>  "module": "fesm2022/ngx-bottom-sheet-modal.mjs",
>   "typings": "index.d.ts",
>   "exports": {
>     "./package.json": {
>       "default": "./package.json"
>     },
>     ".": {
>       "types": "./index.d.ts",
>       "default": "./fesm2022/ngx-bottom-sheet-modal.mjs"
>     }
>   }
> ```
> 
> This is now fixed in `4.0.2` and you project works fine.

Where did you see it working? I'm still getting the same
